Add DM WebUI doc (#7883)
pingcap/docs: dm/dm-webui-guide.md

---

> title: Use WebUI to manage DM migration tasks.

Author: ```suggestion
title: Use WebUI to Manage DM migration tasks.
```

summary: Learn how to use WebUI to manage DM migration tasks.
---

# Use WebUI to manage DM migration tasks

DM WebUI is web-based GUI platform for managing TiDB Data Migration (DM) tasks. It provides a simple and intuitive way to manage a large number of migration tasks, which frees you from using the dmctl command-line tool.

> This document introduces how to access DM WebUI, the prerequisites, the use cases of each interface, and the attention points.

Reviewer: ```suggestion
This document introduces how to access DM WebUI, the prerequisites, the use cases of each page on the interface, and the attention points.
```

> This document introduces how to access DM WebUI, the prerequisites, the use cases of each interface, and the attention points.

Reviewer: ```suggestion
This document introduces how to access DM WebUI, the prerequisites, the use cases of each page on the interface, and the attention points.
```
```suggestion
This document introduces how to access DM WebUI, the prerequisites, the use cases of each interface, and the attention points.
```

> **Warning:**
>
> - DM WebUI is currently an experimental feature. It is not recommended to use it in the production environment.
> - The lifecycle of `task` in DM WebUI has changed, and it is not recommended to use DM WebUI and dmctl at the same time.

DM WebUI has the following pages:

- **Dashboard**: Displays the main monitoring information and status information of migration tasks in DM to help you quickly learn the overall running status of migration tasks and the key metrics on latency and performance.
- **Migration**
    - **Task**: Provides an entry to task creation, and displays the detailed information of each migration task. This page helps you monitor, create, delete, and configure migration tasks.

>     - **Source**: Configures the information of upstream data source for a migration task. On this page, you can manage the upstream configuration in a data migration environment, including creating and deleting upstream configuration, monitoring the task status corresponding to the upstream configuration, and modifying upstream configuration.

Reviewer 2: ` the task status corresponding to the upstream configuration`？ it only allows the user to `enable` and `disable` upstream (or source)

Author: This is a literal translation of 【管理数据迁移环境中的上游配置，包含了，新增、删除上游配置、监控上游配置对应的同步任务状态、修改上游配置等相关的管理功能】. If there is something wrong, could you modify [the Chinese](https://github.com/pingcap/docs-cn/pull/8599/files#diff-9a04649e7332726342eb3b3ea70739397d6a36a2e8c6b17a11355109ca18ad16R22)? Thanks~

>     - **Replication Detail**: Displays the detailed status information of migration tasks. On this page,  you can view the detailed configuration and status information based on a specified filter, including the configuration information and database names of the upstream and downstream.

Reviewer 2: `including the configuration information and database names of the upstream and downstream.`

the users can view the actually replication details here. like the relation of source table and target table.

Author: This is also a literal translation of the Chinese. Based on your comment, I've modified the Chinese: https://github.com/pingcap/docs-cn/pull/8599/files#r831900928. PTAL. Thanks~
```suggestion
    - **Replication Detail**: Displays the detailed status information of migration tasks. On this page,  you can view the detailed configuration and status information based on a specified filter, including the configuration information and database names of the upstream and downstream, the relation of source tables and target tables.
```

+ Enable the DM OpenAPI configuration:

    - If your DM cluster has been deployed using binary, enable the `openapi` configuration item in the configuration of the master node:

        ```
        openapi = true
        ```

    - If your DM cluster has been deployed using TiUP, add the following configuration to the topology file:

        ```yaml
        server_configs:
          master:
            openapi: true
        ```

+ When deploying Grafana for the first time, make sure that the `monitoring_servers` and `grafana_servers` components have been correctly installed. You can expected to configure `grafana_servers` as follows:

    ```
    grafana_servers:
      - host: 10.0.1.14
        # port: 3000
        # deploy_dir: /tidb-deploy/grafana-3000
        config:       # Make sure that the TiUP version for tiup dm -v is later than v1.9.0.
          auth.anonymous.enabled: true
          security.allow_embedding: true
    ```

    If the IP and port of `grafana_servers` are not the default ones, you need to fill in the correct IP and port in the **Dashboard** interface.

> + If your DM cluster is upgraded from an old version, you need to manually modify the Grafana configuration:

Author: ```suggestion
+ If your DM cluster is upgraded from an earlier version, you need to manually modify the Grafana configuration:
```

    1. Edit the `/{deploy-dir}/grafana-{port}/conf/grafana.ini` file as follows:

        ```ini
        [auth.anonymous]
        enabled = true

        [security]
        allow_embedding = true
        ```

    2. Run `tiup dm reload` to make the new configuration effective.

## Dashboard

To see the monitoring of migration tasks, visit the **Dashboard** page. **Dashboard** is an embedded Grafana Dashboard that contains `Standard` and `Professional` views, each of which displays monitoring information from a standard perspective, or from a more professional perspective.

## Migration

**Migration** includes **Source**, **Task**, and **Replication Detail** pages.

## Source

> Before creating a migration task, you need to create the data source information of the upstream for the replication task. You can create the upstream task configuration in the **Source** page. When creating tasks, pay attention to the following items:

Reviewer 2: ```suggestion
Before creating a migration task, you need to create the data source information of the upstream for the replication task. You can create the upstream configuration in the **Source** page. When creating sources, pay attention to the following items:
```

> - If there is a switch between primary and secondary node, enable GTID in the upstream MySQL and set GTID to `True` when creating the upstream configuration; otherwise, the migration task will be interrupted during the switch (except for AWS Aurora).

Reviewer 2: ```suggestion
- If there is a auto failover between primary and secondary instance, enable GTID in the upstream MySQL and set GTID to `True` when creating the upstream configuration; otherwise, the migration task will be interrupted during the failover (except for AWS Aurora).
```

- When multiple migration tasks use the same upstream, it might cause additional stress. Enabling relay log can reduce the impact on the upstream, and it is recommended to enable relay log.

### Task

You can view the migration task details on the **Task** page, and create migration tasks.

#### View migration task details

In the task list, click the task name to view the details page from the right. The details page displays more detailed task status information. In the details page, you can view the status of each sub-task and the current configuration information of the migration task.

In DM, each sub-task of a migration task might be at different stages, namely full dump -> full import (load) -> incremental replication (sync). Therefore, the statistics of a task status and the sub-task statuses can help you better understand the running status of the task.

#### Create migration tasks

To create a migration task on this page, click the **Add** button on the top right corner. You can use one of the following methods to create a migration task:

> - By following the WebUI instruction. Fill in the required information step by step on the WebUI, and this method is suitable for beginners and for daily use.

Reviewer: ```suggestion
- By following the WebUI instruction. Fill in the required information step by step on the WebUI. This method is suitable for beginners and for daily use.
```

> - By using configuration file. Paste or write a JSON-formatted configuration file to create a migration task. This method support adjusting more parameters and is suitable for more advanced users.

Reviewer: ```suggestion
- By using a configuration file. Paste or write a JSON-formatted configuration file to create a migration task. This method supports adjusting more parameters and is suitable for advanced users.
```
```suggestion
- By using configuration file. Paste or write a JSON-formatted configuration file to create a migration task. This method support adjusting more parameters and is suitable for more advanced users.
```

> - By using configuration file. Paste or write a JSON-formatted configuration file to create a migration task. This method support adjusting more parameters and is suitable for more advanced users.

Reviewer: ```suggestion
- By using a configuration file. Paste or write a JSON-formatted configuration file to create a migration task. This method supports adjusting more parameters and is suitable for advanced users.
```

## Replication detail

You can view the status of the migration rules configured for a migration task on the **Replication Detail** page. This page supports querying by task, source, and database name.

The query result contains the corresponding information of the upstream table and the downstream table, so be careful using `.*` in case that too many query results slow down the page response.

## Cluster

### Members

> The **Members** page displays all the master and worker nodes in the DM cluster, and the binding relationship between worker nodes and the source. You can perform simple online or offline operations on master and worker nodes on this page.

Reviewer 2: ```suggestion
The **Members** page displays all the master and worker nodes in the DM cluster, and the binding relationship between worker nodes and the source. 
```
